components/cli/cmd/cellery/cellery.go

```diff
@@ -83,12 +85,25 @@ func newCliCommand(cli cli.Cli, kubeCli kubernetes.KubeCli) *cobra.Command {
 func main() {
 	celleryCli := cli.NewCelleryCli()
 	celleryKubeCli := kubernetes.NewCelleryKubeCli()
+	var ballerinaExecutor ballerina.BalExecutor
+	moduleMgr := &util.BLangManager{}
+	execPath, err := ballerina.GetBallerinaExecutablePath()
+	if err != nil {
+		util.ExitWithErrorMessage("Failed to get executable path", err)
+	}
```

Comment:
Can't we move this logic to NewLocalBalExecutor function? and return an error?

Reply:
Created an interface function "ExecutablePath".